### PR TITLE
Don't require Verilator binary for testing

### PR DESCRIPTION
**Pull request by toddstrader**
Verilator is different from other simulators in that the tool install is not required past the build stage.  Requiring the `verilator` binary to be present when testing presents challenges, especially in distributed  compute settings.

I'm not exactly sure how to test this in CI, but it's getting me past my current problem locally.

### CODE REVIEW

**Comment by toddstrader** [comment]
> I'm not exactly sure how to test this in CI

Let me rephrase.  I can imagine a script which builds a Verilator target, removes Verilator from `$PATH` and then runs the test.  However, that sounds painful / annoying to write especially since there will be likely be differences between my setup and the GitHub CI which would make it necessary to debug via GitHub itself.

So maybe a better way to put this is that I don't believe we're currently testing negative cases for `_simulator_in_path()` and that I'm hopeful that this change is logical / straightforward enough to warrant the same approach.

**Review by themperek — APPROVED** [review]
Looks ok to me to be this way.
